### PR TITLE
Send an email when an organiation invite is created

### PR DESCRIPTION
**Pull request by vishaldeepak**
# What's in this PR?
#520  - Send email when organization invite is created

- Created a email template, `organization_invite_email.ex` file.
- Added test for `organization_invite_email.ex` file.
- Updated organisation invite controller `create` action to send email
- Updated test of organisation invite controller for testing if email is sent
- Updated the doc for the above change

_A bit of confusion on what the redirect url of the email should be._

### CODE REVIEW

**Comment by joshsmith** [comment]
This is looking pretty great! I'll need to check it out locally myself but really love the work you're doing here. 💯

**Comment by vishaldeepak** [comment]
any updates?. Id like to pick up the next  issue 522.

**Comment by joshsmith** [comment]
@vishaldeepak will review shortly. Sorry for the wait!

File: blueprint/api.apib [hunk 1]
```diff
@@ -515,6 +515,8 @@ This endpoint is for Creating, Updating, returning Organization Invites on Code
 
 Admin creates new organization invites which contains `email` and `title`. A `code` is automatically generated on creation and `fulfilled` is set to false.
 
+On successful creation an email is sent to the give `email` address.
```
Comment (joshsmith) [inline blueprint/api.apib]: "give" => "given"

File: config/dev.exs [hunk 2]
```diff
@@ -62,7 +62,8 @@ config :code_corps, CodeCorps.Mailer,
 config :code_corps,
   postmark_forgot_password_template: "123",
   postmark_project_acceptance_template: "123",
-  postmark_receipt_template: "123"
+  postmark_receipt_template: "123",
+  organization_invite_email_template: "123"
```
Comment (joshsmith) [inline config/dev.exs]: In all of these there's a prefix of `postmark_`. We should keep that trend, and alpha-order accordingly.

File: lib/code_corps/emails/organization_invite_email.ex [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      %{
+        title: organization_invite.title,
+        url: url(organization_invite.title, organization_invite.code),
+        subject: "Create your first project on Code Corps"
+      }
+    end
+  
+    defp url(title, code) do
+      Application.get_env(:code_corps, :site_url)
+      |> URI.merge("/organizations/new" <> "?" <> set_params(code, title))
```
Comment (joshsmith) [inline lib/code_corps/emails/organization_invite_email.ex]: I think that `invites/organization` is going to be better here.

File: lib/code_corps/emails/organization_invite_email.ex [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    end
+
+    defp set_params(code, title) do
+      %{code: code, organization_title: title } 
+      |> URI.encode_query
+    end
+  
+    defp template_id, do: Application.get_env(:code_corps, :organization_invite_email_template)
+end
+
```
Comment (joshsmith) [inline lib/code_corps/emails/organization_invite_email.ex]: Weird line break here.

File: lib/code_corps_web/controllers/organization_invite_controller.ex [hunk 5]
```diff
@@ -3,8 +3,7 @@ defmodule CodeCorpsWeb.OrganizationInviteController do
   use JaResource
 
   import CodeCorps.Helpers.Query, only: [id_filter: 2]
-
-  alias CodeCorps.OrganizationInvite
+  alias CodeCorps.{OrganizationInvite, Emails, Mailer}
```
Comment (joshsmith) [inline lib/code_corps_web/controllers/organization_invite_controller.ex]: Would alpha-order these aliases.

File: test/lib/code_corps_web/controllers/organization_invite_controller_test.exs [hunk 6]
```diff
@@ -44,6 +53,12 @@ defmodule CodeCorpsWeb.OrganizationInviteControllerTest do
     test "renders 401 when unauthenticated", %{conn: conn} do
       assert conn |> request_create |> json_response(401)
     end
+
+    # @tag authenticated: :admin
+    # test "sends organization invite email on successful creation", %{conn: conn} do
+    #   organization_invite = request_create(conn, @valid_attrs)
+    #   assert_delivered_email Emails.OrganizationInviteEmail.create(organization_invite)
+    # end
```
Comment (joshsmith) [inline test/lib/code_corps_web/controllers/organization_invite_controller_test.exs]: These lines look like they can be removed.

**Review by joshsmith — CHANGES_REQUESTED** [review]
Hey @vishaldeepak, here are some minor changes I'd like to see. Overall looks really good. I also think we might be missing a plain-text template here.

**Comment by vishaldeepak** [comment]
@JoshSmith what did you mean by `missing a plain text template`? Made other changes!!

**Comment by vishaldeepak** [comment]
@JoshSmith Made changes, let me know if any other change is required. !!

**Comment by joshsmith** [comment]
Do the other emails not have a text-only version? i.e. no HTML?

**Comment by vishaldeepak** [comment]
@JoshSmith Im guessing these files contain the same text as in the html versions. If that is the case then I'm unable to find any. For example I tried searching for familiar text in `forgot_password.html` but can't find any reoccurrence of the text in any other file.

**Comment by begedin** [comment]
@JoshSmith While you can specify a text only version of the postmark template, we do not have any reference to those in the code base, only HTML. I'm not sure if they are specified on the postmark API account.

**Comment by joshsmith** [comment]
We'll need to verify this, then.

**Comment by joshsmith** [comment]
Can you add the following:

- to `config/dev.exs` - `postmark_organization_invite_email_template: 123`
- to `config/staging.exs` - `postmark_organization_invite_email_template: 3442401`
- to `config/prod.exs` - `postmark_organization_invite_email_template: 3441863`

I also realized that `title` is not quite right here. We need to alter the `title` attribute in `OrganizationInvite` to `organization_name`. In the email template, we should also probably be more explicit and change `url` to `invite_url`.

Apologies it took so long to get you this feedback @vishaldeepak.

**Comment by joshsmith** [comment]
Also, it turns out that plain text emails are entirely on Postmark right now. I've apparently hand-jammed them there. Perhaps we should open an issue to also track them in plain-text in our repo?

**Comment by vishaldeepak** [comment]
Made changes, and rebased with develop

**Review by joshsmith — APPROVED** [review]
Looks good. Just a heads up I had meant to change the `title` in the model to `organization_name`. But all good! I've made the changes and we're good to go.

Thanks! 🙌

**Pull request merged**